File: slack-poster/src/test/java/dev/number6/slackposter/generate/DoubleAttributeValueGenerator.java
package dev.number6.slackposter.generate;

import com.amazonaws.services.dynamodbv2.model.AttributeValue;
import uk.org.fyodor.generators.Generator;
import uk.org.fyodor.generators.RDG;
import uk.org.fyodor.range.Range;

public class DoubleAttributeValueGenerator implements Generator<AttributeValue> {

    private final Range<Double> range;

    public DoubleAttributeValueGenerator(double max) {
        this.range = Range.closed(0d, max);
    }

    public DoubleAttributeValueGenerator(Range<Double> range) {
        this.range = range;
    }

    @Override
    public AttributeValue next() {
        return new AttributeValue().withN(RDG.doubleVal(range).next().toString());
    }
}

File: slack-poster/src/test/kotlin/dev/number6/slackposter/generate/DoubleAttributeValueGenerator.kt
package dev.number6.slackposter.generate

import com.amazonaws.services.dynamodbv2.model.AttributeValue
import uk.org.fyodor.generators.Generator
import uk.org.fyodor.generators.RDG
import uk.org.fyodor.range.Range

class DoubleAttributeValueGenerator : Generator<AttributeValue> {
    private val range: Range<Double>

    constructor(max: Double) {
        range = Range.closed(0.0, max)
    }

    constructor(range: Range<Double>) {
        this.range = range
    }

    override fun next(): AttributeValue {
        return AttributeValue().withN(RDG.doubleVal(range).next().toString())
    }
}